model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorage.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.persistence.EntityManager;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.map.common.AbstractEntity;
+import org.keycloak.models.map.storage.MapKeycloakTransaction;
+import org.keycloak.models.map.storage.MapStorage;
+import org.keycloak.models.map.storage.jpa.client.JpaClientMapKeycloakTransaction;
+
+public class JpaMapStorage<V extends AbstractEntity, M> implements MapStorage<V, M> {
```

Comment:
Would this be better positioned as an inner class of `JpaMapStorageProvider`?

Reply:
As discussed it is replaced by anonymous class in `JpaMapStorageProvider`